### PR TITLE
Added support for video cutscenes in polymod using `hxvlc`.

### PR DESCRIPTION
**Pull request by MAJigsaw77**
By using `hxvlc`, `bitstream` playback is available, allowing video cutscenes to be played directly from byte streams. This enables video cutscenes to be played from polymod.

### CODE REVIEW

**Review by Sword352 — APPROVED** [review]
goated.

**Review by CharlesCatYT — APPROVED** [review]

**Review by KarimAkra — APPROVED** [review]

File: source/funkin/ui/title/AttractState.hx [hunk 1]
```diff
@@ -73,9 +73,15 @@ class AttractState extends MusicBeatState
...
-
       add(vid);
-      vid.play(filePath, false);
+
+      openfl.Assets.loadBytes(filePath).onComplete(function(bytes:openfl.utils.ByteArray):Void
+      {
+        if (vid.load(bytes))
+          vid.play();
+
+        onVideoStarted.dispatch();
```
Comment (NotHyper-474) [inline source/funkin/ui/title/AttractState.hx:83]: Looks good, you just forgot to remove this line you copied from VideoCutscene

**Comment by NotHyper-474** [comment]
Also from what I tested this seems to have broken the videos, there's only a black screen instead. But perhaps I did something wrong...

File: source/funkin/ui/title/AttractState.hx (hunk 1 above)
Comment (MAJigsaw77) [reply source/funkin/ui/title/AttractState.hx:83]: > Looks good, you just forgot to remove this line you copied from VideoCutscene

My bad

**Comment by MAJigsaw77** [comment]
> Also from what I tested this seems to have broken the videos, there's only a black screen instead. But perhaps I did something wrong...
> 
> Edit: Ok I've added an error handler to see what was happening and this is what is being thrown. image

The paths seem to have something broken.

**Comment by MAJigsaw77** [comment]
@NotHyper-474 your issue should be gone now as the paths are no longer striped.

**Comment by NotHyper-474** [comment]
> @NotHyper-474 your issue should be gone now as the paths are no longer striped.

Very goob, I'll see if it works tomorrow

**Review by NotHyper-474 — APPROVED** [review]

**Review by AbnormalPoof — APPROVED** [review]
Video cutscenes were actually already supported in Polymod, but you needed to specify the absolute path to the MP4 since Paths() didn't work (see #2446).

PR looks good to me, though!

**Comment by MAJigsaw77** [comment]
> Video cutscenes were actually already supported in Polymod, but you needed to specify the absolute path to the MP4 since Paths() didn't work (see #2446).
> 
> PR looks good to me, though!

Zip mods couldn't load videos in any way, now if these changes are pushed that will be a no problem.

**Comment by AbnormalPoof** [comment]
I see, interesting.

**Review by biomseed — APPROVED** [review]
Pretty sure I heard once that hxvlc is much better than hxCodec so this is good

**Review by cyn0x8 — APPROVED** [review]
yes please

**Comment by ninjamuffin99** [comment]
will have to do R&D on this internally and do some battle testing! but been seeing you work away at hxvlc while we been on old hxcodec !

**Comment by ninjamuffin99** [comment]
on my macbook (havent been at my home windows pc yet) I seem to be getting 
```
Error while executing function UghSong.startVideo()#34: 
An unknown error occurred: hscriptClass(assets/scripts/songs/ugh.hxc):34: Script threw an exception: 
[hxvlc.util.Handle] ERROR: Failed to initialize the LibVLC instance
```
this also occurs with darnell, i just tried to open week 7 and weekend 1 from story menu

**Comment by MAJigsaw77** [comment]
> on my macbook (havent been at my home windows pc yet) I seem to be getting
> 
> ```
> Error while executing function UghSong.startVideo()#34: 
> An unknown error occurred: hscriptClass(assets/scripts/songs/ugh.hxc):34: Script threw an exception: 
> [hxvlc.util.Handle] ERROR: Failed to initialize the LibVLC instance
> ```
> 
> this also occurs with darnell, i just tried to open week 7 and weekend 1 from story menu

You need to do a clean build to delete the old plugins

**Comment by MAJigsaw77** [comment]
I updated libvlc recently but dw, i made a patch were it will automatically fix this issue.

**Comment by MAJigsaw77** [comment]
> I updated libvlc recently but dw, i made a patch were it will automatically fix this issue.

I'll need to release it tomorrow.

**Comment by AbnormalPoof** [comment]
FYI The repo now uses `project.hxp` instead of `Project.xml`, if you're wondering why there's merge conflicts that's likely the reason why

**Comment by cyn0x8** [comment]
nooo why closed? i was really looking forward to this one as it fixes several issues ive encountered with videos

**Comment by MAJigsaw77** [comment]
> nooo why closed? i was really looking forward to this one as it fixes several issues ive encountered with videos

I don't think this was ever going to be pushed, fell free to make a pr on your own if you want.

**Comment by AbnormalPoof** [comment]
To be fair there are a lot of older PRs that still have the `status: pending triage` label. It might take some time ~~especially considering this replaces an entire video library~~.

**Comment by AbnormalPoof** [comment]
> nooo why closed? i was really looking forward to this one as it fixes several issues ive encountered with videos

`hxvlc` appears to have been implemented in v0.6.0 according to [this comment by Eric](https://github.com/FunkinCrew/Funkin/pull/3741#issuecomment-2599174384).

**Pull request closed without merging**